src/librustc/ty/normalize_erasing_regions.rs

```diff
@@ -57,7 +57,9 @@ impl<'tcx> TyCtxt<'tcx> {
     where
         T: TypeFoldable<'tcx>,
     {
-        assert!(!value.needs_subst());
+        if !self.sess.opts.debugging_opts.polymorphize {
+            assert!(!value.needs_subst());
+        }
```

Comment:
This function even takes a `ParamEnv`, so the `needs_subst` check is probably outdated.
cc @nikomatsakis

Reply:
Fixed in 5b3c02b.